addons/pos_adyen/static/src/app/payment_adyen.js

```diff
@@ -314,32 +201,94 @@ export class PaymentAdyen extends PaymentInterface {
             if (line) {
                 line.set_payment_status("force_done");
             }
-
-            return Promise.resolve();
+            return false;
         } else {
             line.set_payment_status("waitingCard");
-            return this.start_get_status_polling();
+            return this.waitForPaymentConfirmation();
         }
     }
 
-    start_get_status_polling() {
-        var self = this;
-        var res = new Promise(function (resolve, reject) {
-            // clear previous intervals just in case, otherwise
-            // it'll run forever
-            clearTimeout(self.polling);
-            self._poll_for_response(resolve, reject);
-            self.polling = setInterval(function () {
-                self._poll_for_response(resolve, reject);
-            }, 5500);
+    waitForPaymentConfirmation() {
+        return new Promise((resolve) => {
+            this.paymentLineResolvers[this.pending_adyen_line().cid] = resolve;
         });
+    }
 
-        // make sure to stop polling when we're done
-        res.finally(function () {
-            self._reset_state();
+    /**
+     * This method is called from pos_bus when the payment
+     * confirmation from Adyen is received via the webhook.
+     */
+    async handleAdyenStatusResponse() {
+        const notification = await this.env.services.orm.silent.call(
+            "pos.payment.method",
+            "get_latest_adyen_status",
+            [[this.payment_method.id]]
+        );
+
+        if (!notification) {
+            this._handle_odoo_connection_failure();
```

Comment:
```suggestion
            this._handle_odoo_connection_failure();
            return;
```

should return immediately right?